src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -115,4 +115,9 @@ private function getClass($value)
 
         return ltrim(is_object($value) ? get_class($value) : $value, '\\');
     }
+
+    protected function createMetadataClass($class)
```

Comment:
Is that ok ?
   
    /**
     * Creates a metadata class
     *
     * @param string $class
     *
     * @return ClassMetadataInterface
     */
    protected function createMetadataClass($class)
    {
        if (!class_exists($class) && !interface_exists($class)) {
            throw new InvalidArgumentException(sprintf('The class or interface "%s" does not exist.', $class));
        }
    
        return new ClassMetadata($class);
    }